### PR TITLE
ENG-907: making AC to watch changes on localStorage and update the locale accordingly

### PR DESCRIPTION
**Pull request by gReis89**

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **kerruba**

If they are not already assigned, you can assign the PR to them by writing `/assign @kerruba` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-admin-console).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/entando-admin-console/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/main/webapp/WEB-INF/apsadmin/jsp/common/tiles-inserts/header.jsp [hunk 1]
```diff
@@ -17,9 +17,19 @@
...
     jQuery(function () {
         $('#activity-stream [data-toggle="tooltip"]').tooltip({trigger: 'hover'});
+
+        // trigger updateLocale if language is changed
+        $('#languageDropdownComponent a').click(function(e) {
+            e.preventDefault();
+            if (updateLocalStorageWithLocale) {
+                updateLocalStorageWithLocale($(this).data('locale'));
+            }
+            window.location.href = $(this).attr('href');
```
Comment (gidesan) [inline src/main/webapp/WEB-INF/apsadmin/jsp/common/tiles-inserts/header.jsp:27]: is this always needed?

File: src/main/webapp/WEB-INF/apsadmin/jsp/common/tiles-inserts/menuAppBuilderIntegration.jsp [hunk 2]
```diff
@@ -17,6 +17,42 @@
...
+        var newUrl = window.location.href;
+        var paramIndex = newUrl.indexOf('request_locale=');
+        if (paramIndex >= 0) {
+            // if the param already exist, replace it with new value
+            var toRemove = newUrl.slice(paramIndex, paramIndex+17);
+            newUrl = newUrl.replace(toRemove, 'request_locale=' + reduxLocale);
+            window.location.href = newUrl;
+        } else {
+            // else only update the url with new param
+            window.location.search = '&request_locale=' + reduxLocale;
```
[the remark this answers is not in the document]
Comment (gReis89) [reply src/main/webapp/WEB-INF/apsadmin/jsp/common/tiles-inserts/menuAppBuilderIntegration.jsp:44]: ops.. I forgot to change the operator, it should be:

```
window.location.search += '&request_locale=' + reduxLocale;
```

File: src/main/webapp/WEB-INF/apsadmin/jsp/common/tiles-inserts/header.jsp (hunk 1 above)
Comment (gReis89) [reply src/main/webapp/WEB-INF/apsadmin/jsp/common/tiles-inserts/header.jsp:27]: we need to intercept the link that changes the language, and call `updateLocalStorageWithLocale` to sync the localStorage with the new language, so if users changes to AP, the new language will be already set.

`updateLocalStorageWithLocale` will only be declared in `menuAppBuilderIntegration.jsp` which is flag-protected, so for other cases, the line 24 will be false and the localStorage won't be updated.

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/entando-admin-console/pull/20#pullrequestreview-452458040" title="Approved">gidesan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-admin-console).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/entando-admin-console/blob/master/OWNERS)~~ [gidesan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**